### PR TITLE
Run clippy with optional features

### PR DESCRIPTION
**Pull request by tustvold**
Runs clippy in CI with optional features enabled, and fixes the issues this exposed

### CODE REVIEW

File: datafusion/common/src/pyarrow.rs [hunk 1]
```diff
@@ -94,19 +94,14 @@ mod tests {
                 let python_path: Vec<&str> =
                     locals.get_item("python_path").unwrap().extract().unwrap();
 
-                Err(err).expect(
-                    format!(
-                        "pyarrow not found\nExecutable: {}\nPython path: {:?}\n\
+                panic!("pyarrow not found\nExecutable: {}\nPython path: {:?}\n\
```
Comment (tustvold) [inline datafusion/common/src/pyarrow.rs:97]: This error appears to have just been being lost previously, which is probably not what was intended

File: datafusion/core/src/scheduler/mod.rs [hunk 2]
```diff
@@ -221,12 +221,12 @@ fn spawn_local_fifo(task: Task) {
 }
 
 #[derive(Debug, Clone)]
-pub struct Spawner {
+pub(crate) struct Spawner {
```
Comment (tustvold) [inline datafusion/core/src/scheduler/mod.rs:224]: This wasn't ever meant to be public, it was accidentally made public as a result of moving from a crate-private module. There was no way to actually construct it, so this is not an API change

File: datafusion/core/src/scheduler/task.rs [hunk 3]
```diff
@@ -35,7 +35,7 @@ use std::sync::{Arc, Weak};
 use std::task::{Context, Poll};
 
 /// Spawns a [`PipelinePlan`] using the provided [`Spawner`]
-pub fn spawn_plan(plan: PipelinePlan, spawner: Spawner) -> ExecutionResults {
+pub(crate) fn spawn_plan(plan: PipelinePlan, spawner: Spawner) -> ExecutionResults {
```
Comment (tustvold) [inline datafusion/core/src/scheduler/task.rs:38]: This is a private module and so this change is practically unnecessary, but needed to placate clippy

File: ci/scripts/rust_clippy.sh [hunk 4]
```diff
@@ -18,4 +18,4 @@
 # under the License.
 
 set -ex
-cargo clippy --all-targets --workspace -- -D warnings
+cargo clippy --all-targets --workspace --features avro,jit,pyarrow,scheduler -- -D warnings
```
Comment (alamb) [inline ci/scripts/rust_clippy.sh:21]: I tried running with 

```suggestion
cargo clippy --all-targets --workspace --all-features -- -D warnings
```


```
error[E0554]: `#![feature]` may not be used on the stable release channel
   --> /Users/alamb/.cargo/registry/src/github.com-1ecc6299db9ec823/packed_simd_2-0.3.8/src/lib.rs:214:1
    |
214 | / #![feature(
215 | |     adt_const_params,
216 | |     repr_simd,
217 | |     rustc_attrs,
...   |
224 | |     custom_inner_attributes,
225 | | )]
    | |__^
```

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @tustvold

File: ci/scripts/rust_clippy.sh (hunk 4 above)
Comment (tustvold) [reply ci/scripts/rust_clippy.sh:21]: Sadly there doesn't appear to be an except feature flag

**Pull request merged**